core/extensions.js

```diff
@@ -83,14 +83,14 @@ Blockly.Extensions.registerMixin = function(name, mixinObj) {
  * decompose are defined on the mixin.
  * @param {string} name The name of this mutator extension.
  * @param {!Object} mixinObj The values to mix in.
- * @param {(function())=} opt_helperFn An optional function to apply after mixing
- *     in the object.
+ * @param {(function())=} opt_helperFn An optional function to apply after
+ *     mixing in the object.
  * @param {Array.<string>=} opt_blockList A list of blocks to appear in the
  *     flyout of the mutator dialog.
  * @throws {Error} if the mutation is invalid or can't be applied to the block.
  */
-Blockly.Extensions.registerMutator = function(name, mixinObj, opt_helperFn,
-    opt_blockList) {
+Blockly.Extensions.registerMutator =
```

Comment:
@NeilFraser fyi this causes eslint errors.  It's fine with 
```
Blockly.Extensions.checkHasFunction_ = function(errorPrefix, func,
    propertyName) {
  if (!func) {
``` 
but breaks on 
```
Blockly.Extensions.checkHasFunction_ =
    function(errorPrefix, func, propertyName) {
  if (!func) {
```

Reply:
Interesting.  That's in direct violation of our style guide:
https://google.github.io/styleguide/jsguide.html#formatting-where-to-break

Reply:
It's in approximate violation, since the style guide says that there is no comprehensive formula.  

I want to use the version that leaves `function` on the first line because it means we don't need to put eslint exceptions everywhere. 

If we can get rid of all eslint errors and turn linting on in travis, we will have cleaner code overall, and reduce time spent pointing out style issues in review.  I think that's worth using a slightly different line-break strategy here.